src/main/java/io/aiven/kafka/connect/s3/S3SinkConfig.java

```diff
@@ -109,13 +113,41 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(S3SinkConfig.class);
 
 
-    public S3SinkConfig(final Map<String, String> originals) {
-        super(configDef(), originals);
+    public S3SinkConfig(final Map<String, String> properties) {
+        super(configDef(), handleDeprecatedYyyyUppercase(properties));
```

Comment:
[ Super minor] I would suggest
```suggestion
        super(configDef(), handleDeprecatedYyyyUppercase(properties) if self.use_deprecated_yyyy else properties);
        
       def use_deprecated_yyyy(self):
         properties.containsKey(AWS_S3_PREFIX_CONFIG) or properties.containsKey(AWS_S3_PREFIX)
```

Reply:
Otherwise handleDeprecatedYyyyUppercase also responsible for handling cases when we should not use deprecated (even if it mean just to return properties). It could encourage people to handle case for a not deprecated stuff in this function

Reply:
I did it a bit differently by changing the naming of the things.